src/PermissionKernal.java

```diff
@@ -0,0 +1,17 @@
+/**
+ * The permission users need to perform an action.
+ */
+public interface PermissionKernal {
```

Comment:
This sort of skips ahead into part 3, so I'll make my comments as if this is part 3. Naming wise, I believe it's spelled `kernel`. Also, make sure this inherits from `Standard`.